src/NimBLEClient.cpp

```diff
@@ -313,9 +314,36 @@ bool NimBLEClient::connect(const NimBLEAddress& address, bool deleteAttributes,
  * @return True on success.
  * @details This is a blocking function and should not be used in a callback.
  */
-bool NimBLEClient::secureConnection() const {
+bool NimBLEClient::secureConnection(bool async) {
     NIMBLE_LOGD(LOG_TAG, ">> secureConnection()");
 
+    if (!NimBLEDevice::m_synced) {
+        NIMBLE_LOGE(LOG_TAG, "Host reset, wait for sync.");
+        return false;
+    }
```

Comment:
Probably not needed, if the stack get out of sync the client will be disconnected and the call will fail with ENOTCONN